lib/charms/tls_certificates_interface/v1/tls_certificates.py

```diff
@@ -1089,24 +1178,25 @@ def _on_update_status(self, event: UpdateStatusEvent) -> None:
         """
         relation = self.model.get_relation(self.relationship_name)
         if not relation:
+            logger.warning(f"No relation: {self.relationship_name}")
+            return
+        provider_relation_data = _load_relation_data(relation.data[relation.app])  # type: ignore[index]  # noqa: E501
+        if not self._relation_data_is_valid(provider_relation_data):
+            logger.warning(
+                f"Provider relation data did not pass JSON Schema validation: {relation.data[relation.app]}"  # type: ignore[index]  # noqa: W505
+            )
             return
-        for unit in relation.units:
-            relation_data = _load_unit_relation_data(relation.data[unit])
-            if self._relation_data_is_valid(relation_data):
-                certificates = relation_data.get("certificates")
-                if not certificates:
-                    continue
-                for certificate_dict in certificates:
-                    certificate = certificate_dict["certificate"]
-                    certificate_object = x509.load_pem_x509_certificate(data=certificate.encode())
-                    time_difference = certificate_object.not_valid_after - datetime.utcnow()
-                    if time_difference.total_seconds() < 0:
-                        logger.warning("Certificate is expired")
-                        self.on.certificate_expired.emit(certificate=certificate)
-                        self.request_certificate_revocation(certificate)
-                        continue
-                    if time_difference.total_seconds() < (self.expiry_notification_time * 60 * 60):
-                        logger.info("Certificate almost expired")
-                        self.on.certificate_expiring.emit(
-                            certificate=certificate, expiry=certificate_object.not_valid_after
-                        )
+        for certificate_dict in self._provider_certificates:
+            certificate = certificate_dict["certificate"]
+            certificate_object = x509.load_pem_x509_certificate(data=certificate.encode())
+            time_difference = certificate_object.not_valid_after - datetime.utcnow()
```

Comment:
Should we consider using `datetime.now(timezone.utc)` instead of `datetime.utcnow()`? A [warning in the docs](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow) explains why.
> Warning Because naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. As such, the recommended way to create an object representing the current time in UTC is by calling datetime.now(timezone.utc).

Reply:
This doesn't seem to work. The datetime taken from the `load_pem_x509_certificate` is not timezone aware while the one using `datetime.now(timezone.utc)` is and those 2 quantities can't be subsctracted.

Reply:
In this particular case, this will not be an issue, but I think it is still best to follow the recommendation, at least so that we get used to this construct.

Reply:
@ghislainbourgeois Like I mentioned above, I don't see how that's doable in our case.